### PR TITLE
feat: add support for avax chains

### LINKED ISSUE #638: Move `GasOverheads` to chain spec (or hardcode)
**Describe the feature**
`GasOverheads` is currently only used in its default form without any way to configure. We should either move these to hardcoded constants or move this into chain spec.

### PR DESCRIPTION
**Pull request by dancoombs**
Closes #638

## Proposed Changes

- Add gas overheads to chain spec (#638)
- Allow for retries in chain sync
- Add avax and avax fuji chain specs

### CODE REVIEW

File: bin/rundler/chain_specs/avax_fuji.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+base = "avax"
+
+name = "Avax-Fuji"
+id = 43113
+
+# priority_fee_oracle_type = "FIXED"
```
Comment (alex-miao) [inline bin/rundler/chain_specs/avax_fuji.toml:6]: Can we delete these commented out values

File: crates/builder/src/bundle_proposer.rs [hunk 2]
```diff
@@ -1444,13 +1443,14 @@ mod tests {
...
+        let cs = ChainSpec::default();
+
         let expected_gas = math::increase_by_percent(
             op.pre_verification_gas
                 + op.verification_gas_limit * 2
                 + op.call_gas_limit
-                + ov.bundle_transaction_gas_buffer
-                + ov.transaction_gas_overhead,
+                + cs.transaction_intrinsic_gas
+                + 5_000, // v0.6 inner gas overhead
```
Comment (0xfourzerofour) [inline crates/builder/src/bundle_proposer.rs:1453]: should we put this into a const?

**Review by 0xfourzerofour — APPROVED** [review]
Looks good

File: crates/types/src/chain.rs [hunk 3]
```diff
@@ -43,6 +43,21 @@ pub struct ChainSpec {
...
+    pub transaction_intrinsic_gas: U256,
+    /// Per user operation gas cost for v0.6
+    pub per_user_op_v0_6_gas: U256,
+    /// Per user operation gas cost for v0.7
+    pub per_user_op_v0_7_gas: U256,
+    /// Per user operation deploy gas cost overhead, to capture
+    /// deploy costs that are not metered by the entry point
+    pub per_user_op_deploy_overhead_gas: U256,
+    /// Gas cost for a user operation word in a bundle transaction
+    pub per_user_op_word_gas: U256,
```
Comment (alex-miao) [inline crates/types/src/chain.rs:56]: Do these values ever change across chains

Comment (dancoombs) [reply crates/types/src/chain.rs:56]: `per_user_op_deploy_overhead_gas` does (Avax). Currently `per_user_op_word_gas`, `calldata_zero_byte_gas`, and `calldata_non_zero_byte_gas` don't, but a simple EVM config change could change that, so kept them configurable with a reasonable default.

File: crates/builder/src/bundle_proposer.rs (hunk 2 above)
Comment (dancoombs) [reply crates/builder/src/bundle_proposer.rs:1453]: Yup, fixed.

**Review by alex-miao — APPROVED** [review]

**Review by 0xfourzerofour — APPROVED** [review]

**Pull request merged**